### ISSUE TITLE
Authentication with Azure AD

### ISSUE THREAD

**Issue by shars95** [mstaal/msal_streamlit_authentication#3]
Hi mstaal ,

Thank you for this module , it has saved a lot of time for me authenticating with azure AD. /
I have one question though. Why there is no option for "client secret" used to identify the registered application as a trusted application to access the user's information securely. Is there a way to enable this ?

Thanks again!

**Comment by mstaal** [comment]
Hi @shars95 ! 
Sorry for the late reply and thanks for your question. May I ask what your use case is, and why you would to use a client secret? The current design of the plugin is to support the OpenID Authorization Code flow (via PKCE) which does not rely on a client secret. The Client Credentials flow (which is a flow for system accounts) uses secrets, but it would be easy to implement something for that flow specifically.
Am I misinterpreting your question?

And thanks for your feedback! It's nice to know that the plugin is cherished. :-D

**Comment by tipani86** [comment]
This is very useful! I was able to quickly get into the login screen. However, like the OP of this thread, I also saw that my registered app was deemed "untrustworthy" so the user has to go through an extra step to confirm that they want to use it (it's our corporate internal app, so should skip this part, as long as they are logged in through the work account).

In your sample code, there is a comment `# Corresponds to the 'auth' configuration for an MSAL Instance` when specifying the auth dictionary.

Assuming it is that, I added my own key called `clientCredential` that is supposed to correspond to the app secret. However, it doesn't seem it had any effect because I still got asked to confirm the app when I login.

**Comment by mstaal** [comment]
Thanks a lot, @tipani86 ! I appreciate that it serves a purpose for you. Sorry for asking explicitly, but have you now resolved your issue? Or else I will try to look further into it. I try to make small adjustments every now and then.

**Comment by tipani86** [comment]
> Thanks a lot, @tipani86 ! I appreciate that it serves a purpose for you. Sorry for asking explicitly, but have you now resolved your issue? Or else I will try to look further into it. I try to make small adjustments every now and then.

Yeah I mean I just accept the fact that the user has to explicitly click an approval the first time they log in. Otherwise, it's fine.

Perhaps an unrelated issue were sometimes when I log in on different machines and get displayed a logged in status (with the button written as "Logout") but the site still runs as if I am not logged in, i.e. the login token is not present. Are you aware of such sync issues?

I can solve it by clicking Logout and going through the whole logout - login dance.

**Comment by mstaal** [comment]
@tipani86 : Thanks for letting me know. I will try to look at it at some point. Generally, my feeling is that syncing is definitely an obstacle when getting a procedural / synchronous language like Python to sing with JavaScript, which is async in its very nature.
